### ISSUE TITLE
"contextmenu" (right click) event is not implemented

### ISSUE THREAD

**Issue by ChrisFloofyKitsune** [mikke89/RmlUi#595]
I would like to be able to use right click to interact with things- but the go to event for that is not implemented.

https://developer.mozilla.org/en-US/docs/Web/API/Element/contextmenu_event

For now, I don't mind doing a workaround using mousedown/up

**Comment by mikke89** [comment]
Thanks for the suggestion. Certainly sounds like a feature that we could add, I'd happily take a pull request.

I guess the reason this hasn't been implemented yet, is the assumption that users control the input side already, so they could do their actions directly on the library when receiving a right click. I guess a built-in event could add some value by event handling, and playing better with data bindings and script. It would be nice to understand the use case a bit better, could you elaborate on that a bit?

**Comment by ChrisFloofyKitsune** [comment]
The main uses I see for this in what I'm working on (and in general) is to be able to right click usernames in a chat window or a player list- bringing up a small menu for things like Ignore, Whisper, Add Friend, Report, etc

it's a lot more friendly than having a weird alt+mouse button keybind or having to type in a /command into the chat to do the same.

**Comment by mikke89** [comment]
Yeah, I fully see the need for a context menu. I'm just trying to understand which alternatives you have considered for opening them and if they have any limitations. In particular, the library already submits `mousedown` and `mouseup` events also during right-click. Alternatively, the client can always open menus as desired, after all, the client supplies the information to the library about right mouse button being clicked, so they can act on that independently.

**Comment by mikke89** [comment]
Hi, I am closing this one as not planned. It's simply not a high enough priority, considering this one can quite trivially be solved by using the `mouseup` event and looking at the `button` parameter.
